### ISSUE TITLE
"level 3" not working with jsonp files

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/robotframework-testsuitesmanagement#162]
The current "level 3" code is still related to '`json`' files, but not able to find files with new extension '`jsonp`':

```
if not bLevel3Check:
    if os.path.isfile(self.sTestcasePath + 'config' + os.path.sep + 'robot_config.json'):
        self.sTestCfgFile = self.sTestcasePath + 'config' + os.path.sep + 'robot_config.json'
```
Expected: Testsuites Management is able to handle both extensions: `json` and `jsonp`.

To be clarified: What shall happen in case of both files can be found in `config` folder: `robot_config.json` and `robot_config.jsonp`?

Maybe a test stop with error message would be the best choice.

**Comment by test-fullautomation** [comment]
Hi Holger,

level 3 loads automatically files.
If there is more than one json/jsonp file, then it's not clear which file need to be loaded.
Therefore we need an error message like:
Only one file is allowed for automatic configuration (level 3). Found the following files:
path/path/...
path/path/...

Thank you,
 Thomas

**Comment by namsonx** [comment]
Hello Thomas,

I agree with Holger that we should allow the configuration file either end with 'json' or 'jsonp'.
The mechanism of configuration level 3 is:
    - Checking the existing 'config' folder in current folder which robot script is located in.
    - If 'config' folder is existed:
           + If there is any json configuration file have the same name with robot test script, it will use this json file for configuration level 3.
           + If not above and there is 'robot_config.json' file in config folder, it will use the 'robot_config.json' for configuration file for configuration level 3.
           + In these 2 cases above are not matched, then configuration level 4 is enabled.

Thank you,
Son

**Comment by test-fullautomation** [comment]
Hi Son,

I modified your description from above:

The mechanism of configuration level 3 is:

- Checking the existing 'config' folder in current folder which robot script is located in.
- If 'config' folder is existing:
+ If there is a **jsonp** or **json** configuration file has the same name with robot test script, it will use this **jsonp**/**json** file for configuration level 3 **for this test scrip**t.
+ If not above and there is 'r**obot_config.jsonp**' or '**robot_config.jsonp**' file in config folder, it will use the '**robot_config.json/p**' for configuration file for configuration level 3.
+ In these 2 cases above are not matched, then configuration level 4 is enabled.

First check always for jsonp, then json.
If there are jsonp and json files with the same name (before "."), then it's an ambiguous situation. It's not clear which file to use. Therefore we raise in any case a speaking error message like I described above.

Please implement accordingly.

Thank you,
 Thomas

**Comment by HolQue** [comment]
Addendum:

In case of `jsonp` **and** `json`

please consider:

https://github.com/test-fullautomation/robotframework-testsuitesmanagement/issues/162#issuecomment-1547992224

**Comment by HolQue** [comment]
Added lately: `TSM_0065` and `TSM_0066`

https://github.com/test-fullautomation/robotframework-testsuitesmanagement/blob/develop/test/TSM_TestUsecases.rst

**Comment by HolQue** [comment]
Hi Son,

please give the following alternative example code a try. It solves the issue (and is easier to understand):

```
# ============================== alternative version
if not bLevel3Check:
    sJsonFileBase = CString.NormalizePath(f"{self.sTestcasePath}/config/robot_config")
    sJsonFile1    = f"{sJsonFileBase}.jsonp"
    sJsonFile2    = f"{sJsonFileBase}.json" # still supported alternative extension

    if os.path.isfile(sJsonFile1) and os.path.isfile(sJsonFile2):
        errorMessage = "Configuration file duplicate detected (both extensions: 'jsonp' and 'json')!\n" + \
                       f"* file 1: '{sJsonFile1}'\n" + \
                       f"* file 2: '{sJsonFile2}'\n" + \
                       "Please decide which one to keep and which one to remove. Both together are not allowed."
        logger.error(errorMessage)
        BuiltIn().unknown("Configuration file duplicate detected (both extensions: 'jsonp' and 'json')!")
    elif os.path.isfile(sJsonFile1):
        self.sTestCfgFile = sJsonFile1
        self.rConfigFiles.bLevel4 = False
    elif os.path.isfile(sJsonFile2):
        self.sTestCfgFile = sJsonFile2
        self.rConfigFiles.bLevel4 = False
    else: # meaning: if not os.path.isfile(sJsonFile1) and not os.path.isfile(sJsonFile2)
        self.rConfigFiles.bLevel3 = False
        if not self.bConfigLoaded:
            #self.rConfigFiles.bLevel4 = True
            sDefaultConfig=str(pathlib.Path(__file__).parent.absolute() / "robot_config.json")
            self.sTestCfgFile = sDefaultConfig
```

The output in console is:

```
[ ERROR ] Configuration file duplicate detected (both extensions: 'jsonp' and 'json')!
* file 1: 'C:/RobotTest/projects/ROBFW_AIO/robotframework-testsuitesmanagement/test/testfiles/configfoldertests2/config/robot_config.jsonp'
* file 2: 'C:/RobotTest/projects/ROBFW_AIO/robotframework-testsuitesmanagement/test/testfiles/configfoldertests2/config/robot_config.json'
Please decide which one to keep and which one to remove. Both together are not allowed.
Test Case tsm-cft-testfile-2 (BADCASE) :: tsm-cft-testfile-2 (BADC... | UNKNOWN |
Parent suite setup unknown:
Configuration file duplicate detected (both extensions: 'jsonp' and 'json')!

```

**Comment by namsonx** [comment]
Hello Holger,

Thank you for your hint, I updated the pull-request.

Thank you,
Son

**Comment by test-fullautomation** [comment]
solved and already merged with #171
